pandas/core/groupby/ops.py

```diff
@@ -232,6 +232,7 @@ def indices(self):
             label_list = [ping.labels for ping in self.groupings]
             keys = [com.values_from_object(ping.group_index)
                     for ping in self.groupings]
+            keys = [np.array(key) for key in keys]
```

Comment:
this needs to be in get_indexer_dict in pandas/core/sorting.py

also, use ``np.asarray`` (np.array like this copies)